tests/unit/s2n_handshake_test.c

```diff
@@ -35,6 +35,65 @@
 #include "tls/s2n_tls13.h"
 #include "utils/s2n_safety.h"
 
+enum test_type {
+    TEST_TYPE_START,
+    TEST_TYPE_SYNC = TEST_TYPE_START,
+    TEST_TYPE_ASYNC,
+    TEST_TYPE_END
+} test_type;
+
+struct s2n_async_pkey_op *pkey_op = NULL;
+int async_pkey_op_called = 0;
+int async_pkey_op_performed = 0;
+
+static int handle_async(struct s2n_connection *server_conn)
```

Comment:
If it isn't there yet I would recommend also having unit-tests which cover the cancellation scenarios (freeing the Op before executing it, freeing it after it was executed but before calling `s2n_async_pkey_op_apply`, freeing  the connection before the op is called, etc.

Reply:
Yup, working on those.

Reply:
done